### PR TITLE
feat: guild commands & ephemeral interaction callback

### PR DESCRIPTION
**Pull request by oscarfsbs**
No clue what the best way to write the rest of the make_basic_test_reply function is, some help making the rest of the [callbacks](https://discord.com/developers/docs/interactions/receiving-and-responding#interaction-response-object-interaction-callback-data-structure) extensible would be appreciated :>

### CODE REVIEW

**Comment by Cyteon** [comment]
I'll take a look soon

**Comment by ninsaccplayzyt** [comment]
:sippycup:

File: test/example_bot.gleam [hunk 1]
```diff
@@ -177,7 +179,8 @@ fn event_handler(bot, packet: event_handler.Packet) {
 
       case interaction.d.data.name {
         "test" -> {
-          discord_gleam.interaction_reply_message(interaction, "test")
+          discord_gleam.interaction_reply_message(interaction, "test", True)
+          discord_gleam.interaction_reply_message(interaction, "test", False)
```
Comment (Cyteon) [inline test/example_bot.gleam:183]: You can only reply to a interaction once, maybe make two commands test and test2? where one is global and one is guild?

File: test/example_bot.gleam [hunk 2]
```diff
@@ -28,8 +27,11 @@ pub fn main(token: String, client_id: String) {
       ],
     )
 
-  discord_gleam.wipe_slash_commands(bot, client_id)
-  discord_gleam.register_commands(bot, client_id, [test_cmd])
+  discord_gleam.wipe_global_commands(bot, client_id)
+  discord_gleam.register_global_commands(bot, client_id, [test_cmd])
+
+  discord_gleam.wipe_guild_commands(bot, client_id, guild_id)
+  discord_gleam.register_guild_commands(bot, client_id, guild_id, [test_cmd])
```
Comment (Cyteon) [inline test/example_bot.gleam:34]: Here we could probably have two diffrent commands like test and test2

**Review by Cyteon — CHANGES_REQUESTED** [review]
Other than this everything else seems fine

File: test/example_bot.gleam (hunk 1 above)
Comment (ninsaccplayzyt) [reply test/example_bot.gleam:183]: doesnt matter :sippycup:

File: test/example_bot.gleam (hunk 2 above)
Comment (ninsaccplayzyt) [reply test/example_bot.gleam:34]: doesnt matter :sippycup:

**Pull request merged**